app/controllers/api/api_controller.rb

```diff
@@ -393,4 +408,47 @@ def pept2go_helper
 
     output
   end
+
+  def pept2interpro_helper
+    output = {}
+    ipr_entries = []
+
+    @sequences = Sequence.where(sequence: @input)
+
+    @sequences.each do |seq|
+      fa = seq.calculate_fa(@equate_il)
+      iprs = fa['data'].select { |k, _v| k.start_with?('IPR:') }
+
+      output[seq.sequence] = {
+          total: fa['num']['all'],
+          ipr: iprs.map do |k, v|
+            {
+                code: k[4..-1],
+                protein_count: v
+            }
+          end
+      }
+
+      ipr_entries.push *(iprs.map { |k, _v| k[4..-1] })
+    end
+
+    if @extra_info
```

Comment:
@bmesuere It might be interesting to also support some kind of split by domains when users pass an extra parameter, like we do for GO-terms.

Reply:
Added functionality to split InterPro entries into groups according to type in d8ecf82.